p2p/client_test.go

```diff
@@ -119,25 +114,25 @@ func TestGossiping(t *testing.T) {
 
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
-	clients := startTestNetwork(ctx, t, 5, map[int][]int{
-		1: []int{0},
-		2: []int{0},
-		3: []int{1},
-		4: []int{2},
+	clients := startTestNetwork(ctx, t, 5, map[int]hostDescr{
+		1: hostDescr{conns: []int{0}, chainID: "test"},
+		2: hostDescr{conns: []int{0}, chainID: "test"},
+		3: hostDescr{conns: []int{1}, chainID: "test"},
+		4: hostDescr{conns: []int{2}, chainID: "test", realKey: true},
 	}, logger)
 
 	// wait for clients to finish refreshing routing tables
 	clients.WaitForDHT()
 
-	time.Sleep(5 * time.Second)
+	time.Sleep(10*time.Second)
 
 	// gossip from client 4
 	err := clients[4].GossipTx(ctx, []byte("sample tx"))
 	assert.NoError(err)
 
-	nCtx, nCancel := context.WithTimeout(ctx, 1*time.Second)
+	nCtx, nCancel := context.WithTimeout(ctx, 300*time.Second)
```

Comment:
Wowser. Does that mean, the messages gets now written to the channel but the client isn't able to retrieve it now?

Reply:
This is a work in progress, big value to give me more time in debugger ;)

Reply:
Doesn't a breakpoint stop the "world"? Or does that depend on which debugger you are using?